armada_jupyter/submissions.py

```diff
@@ -0,0 +1,205 @@
+
+
+class Submission:
+    """
+    Represents a Armada-Jupyter Submission
+    """
+
+    def __init__(
+        self,
+        name: str,
+        image: str,
+        armada_queue: str,
+        armada_priority: int,
+        timeout: str,
+        resources: Optional[K8sResources] = None,
+    ):
+        self.name = name
+        self.image = image
+        self.armada_queue = armada_queue
+        self.armada_priority = armada_priority
+        self.timeout = self.timeout_conversion(timeout)
+        self.resources = resources
+
+        # Check that resources is the right object
+        if not isinstance(resources, K8sResources) and resources is not None:
+            raise ValueError(
+                f"resources must be a K8sResources object, not {type(resources)}"
+            )
+
+    @staticmethod
+    def timeout_conversion(timeout: str) -> Optional[str]:
+        """
+        Convert timeout from minutes and hours into seconds.
+
+        timeout can be in the form of 10m or 5h for example
+
+        should return a string like 100s
+        """
+
+        if timeout[-1] == "m":
+            return str(int(timeout[:-1]) * 60) + "s"
+        elif timeout[-1] == "h":
+            return str(int(timeout[:-1]) * 60 * 60) + "s"
+        elif timeout[-1] == "s":
+            return timeout
+
+        ValueError("Timeout must be in the form of [x]h, [x]m or [x]s")
+        return None
+
+    def resources_from_dict(
+        self, resources: Dict[str, Dict[str, Union[str, int]]]
+    ) -> None:
+        """
+        Used for adding a resources from dict
+        """
+
+        self.resources = k8s_resources_from_dict(
+            limits=resources.get(YMLSTR.limits),
+            requests=resources.get(YMLSTR.requests),
+        )
+
+    def __repr__(self) -> str:
+        return (
+            f"Submission(f{YMLSTR.name}='{self.name}', {YMLSTR.image}='{self.image}' "
+            f"{YMLSTR.armada_queue}='{self.armada_queue}', "
+            f"{YMLSTR.armada_priority}={self.armada_priority}, "
+            f"{YMLSTR.timeout}='{self.timeout}', {YMLSTR.resources}={self.resources})"
+        )
+
+
+def k8s_resources_from_dict(
+    limits: Optional[Dict[str, Union[str, int]]] = None,
+    requests: Optional[Dict[str, Union[str, int]]] = None,
+) -> K8sResources:
+    """
+    Creates a K8sResources object from a dict
+    """
+
+    changed: Dict[str, Optional[K8sResourceOptions]] = {}
+
+    if limits is None and requests is None:
+        raise ValueError("Must specify at least one of limits or requests")
+
+    # Use a loop to save repeated code.
+    for key, needs_changed in [(YMLSTR.limits, limits), (YMLSTR.requests, requests)]:
+        if needs_changed is None:
+            changed[key] = None
+            continue
+
+        # We use .get() here because we don't want to raise an error if the key
+        # doesn't exist.
+        changed[key] = K8sResourceOptions(
+            cpu=int(needs_changed.get(YMLSTR.cpu, DEFAULT_CPU)),
```

Comment:
I am not 100% where we should place these defaults, whether we do it in the functions or in the classes. Open to opinions.